### DISCUSSION TITLE
Cost comparison problem

### DISCUSSION THREAD

**Discussion by stephencrowther** [BottlecapDave/HomeAssistant-OctopusEnergy#1046] [Q&A] [answered]
Tried to setup a cost comparison comparing my current Agile tariff with the standard flexible octopus E7 tariff.

Agile sensor returns a cost but the other flexible octopus returns the first time code in list of time codes.

not sure if I’m doing anything wrong but I thought this tool just needed to know tariff codes etc and did all the calculations for you?

**Comment by BottlecapDave** [comment]
Can you post a screenshot of what's being returned ( the sensor value and it's attributes) as I'm not quite sure what you mean. Thr sensor should just return the cost. It relies on the same consumption data being available as the normal previous consumption data so might take up to 24 hours to populate.

**Reply by stephencrowther** [reply]
Sure. I’m not ruling out user error here as I’m very new with ha but I filled out the form when creating the sensor as per the instructions.

![IMG_3328](https://github.com/user-attachments/assets/c057b169-dbf8-4baa-a15e-beb688cb250b)
![IMG_3327](https://github.com/user-attachments/assets/5f87932b-fb61-41c6-87a3-9603f589f8a1)
![IMG_3326](https://github.com/user-attachments/assets/dae3f198-afad-46e9-a738-9a3ca2cf8c55)
![IMG_3322](https://github.com/user-attachments/assets/70fb6af3-59c2-40e8-bb34-693c31dd9912)

**Reply by BottlecapDave** [reply, marked as answer]
The data and sensors look fine to me. The second sensor reports the date the rates that are used for the comparison was last updated. This part of the integration is used for comparing your standard cost sensor with the sensor that is provided when configuring the tariff override (the £2.03 in your screenshots). How you compare the two sensors is entirely up to you. I for instance use a stats graph to compare the two sensors so I can see which one is higher over a period.

```
chart_type: line
period: hour
type: statistics-graph
entities:
  - >-
    sensor.octopus_energy_electricity_xxx_xxx_previous_accumulative_cost
  - >-
    sensor.octopus_energy_electricity_xxx_xxx_previous_accumulative_cost_e7flexible
stat_types:
  - state
```

Is this what you were expecting? If not, what were you expecting?

**Reply by stephencrowther** [reply]
Ah so the entity should t show a cost value then like the current tariff one shows 2.03 in that example - I thought both entities were supposed to show a value.

I’ll try the graph as yes that was my next thing to do, I just didn’t think it was setup properly to create a graph because of it not reporting a £ value for the comparison tariff.

sorry if I’ve wasted your time

**Reply by stephencrowther** [reply]
Sorry - I understand what is going on now - those two entities are both the comparison tariff.

For some reason I thought one of those entities was for agile and one for whatever I chose to compare. Of course I already have an agile cost as part of the main integration doh!

sorry about that thanks for kicking me in the right direction to understand!

**Reply by BottlecapDave** [reply]
No problem and glad it's all working :)
